### ISSUE TITLE
Item find_by_id raising 500 exception error

### ISSUE THREAD

**Issue by jordangraft** [ruckus/quickbooks-ruby#274]
With this commit 057d4907fe3546111c94dfccc4845e5480036d5d we are getting an error when we try to ```fetch_by_id``` for an ```Item```.

Request
```
service = Quickbooks::Service::Item.new(initialize_hash)
service.fetch_by_id(product.qb_id)
```

Response
```
Quickbooks::IntuitRequestException: General Authentication Error:AuthenticationErrorGeneral: Internal Server Error , statusCode: 500
```

**Comment by ruckus** [comment]
What does your `initialize_hash` look like? Judging by the error its an authentication issue, so maybe you're not injecting the auth params correctly

**Comment by jordangraft** [comment]
This is the initialize_hash variable ```{company_id: company_id, access_token: access_token}```.

The ```fetch_by_id``` test is one of 80 and its the only one failing so I feel good that it's something with this specific method.  I can also confirm that the tests passes for commits before 057d4907fe3546111c94dfccc4845e5480036d5d.

Happy to help look for the answer if you have some direction on where to start.  And also thanks a bunch for this gem.  It's a lot of work and our team is very appreciative of your efforts!

**Comment by ruckus** [comment]
Interesting. Strange that an auth error would be raised by the seemingly innocuous inclusion of a query-parameter (the `minorversion=4`). But it sounds like that is what is happening. 

On the other hand the `url_for_resource` and `url_for_query` methods are indeed used in basically every call, so if QBO did have an issue with that parameter one would think the error would be happening much more frequently.

How about this: 

* enable debugging: https://github.com/ruckus/quickbooks-ruby#logging
* report back with the full URL and request/response

I think it might help to see the full details. I'm wondering if for instance the URL is not being generated correctly and the auth error is a red-herring

**Comment by jordangraft** [comment]
It's the instance url.  the ```fetch_by_id``` method is calling ```url_for_resource``` and then concatenating the item id on the end but this new method in service/item is adding the minorversion query param ahead of that.

```
      def url_for_resource(resource)
        url = super(resource)
        "#{url}?minorversion=#{Quickbooks::Model::Item::MINORVERSION}"
      end
```

so you are getting something like ```/v3/company/<companyID>/items?minorversion=4/2``` instead of ```/v3/company/<companyID>/items/2?minorversion=4``` 

I can make a PR if you want.

**Comment by ruckus** [comment]
Wonderful. That was my similar thought. Yes, if you could make a PR that would be fantastic. Thanks!

**Comment by minimul** [comment]
Closed in #273
